std/conv.d

```diff
@@ -5024,6 +5024,48 @@ T emplace(T, Args...)(void[] chunk, auto ref Args args)
     return result;
 }
 
+/**
+Constructs an object of $(D class) type $(D T) at the location $(D obj)
+references. The constructor is passed the arguments $(D Args).
+
+$(D obj) must not be a base class or interface.
+
+This function can be $(D @trusted) if the corresponding constructor of
+$(D T) is $(D @safe).
+
+Returns: If $(D obj) is null, it will return a null reference of type $(D T).
+Otherwise, returns a reference to a newly constructed $(D T) at $(D obj)'s location.
+ */
+T emplace(T, Args...)(T obj, auto ref Args args) @trusted
+    if (is(T == class))
+{
+    if (obj is null)
+        return null;
+    if (typeid(T) !is typeid(obj))
```

Comment:
typeid returns the most derived type for an instance, so for example:
```D
class C {}

void main()
{
   Object o = new C;
   assert(typeid(Object) !is typeid(o));
}
```

emplace called with `o` as defined above as the `T` parameter would fail this assert.